lib/paper_trail/version_concern.rb

```diff
@@ -417,7 +417,14 @@ def reify_has_many_through(associations, model, options = {})
       associations.each do |assoc|
         next unless assoc.klass.paper_trail_enabled_for_model?
         through_collection = model.send(assoc.options[:through])
-        collection_keys = through_collection.map { |through_model| through_model.send(assoc.foreign_key) }
+
+        # if the association is a has_many association again, then call reify_has_manys for each through_collection
+        if !assoc.source_reflection.belongs_to? && through_collection.present?
+          through_collection.each { |through_model| reify_has_manys(through_model,options) }
+          return
```

Comment:
> I believe we want to use next instead of return here, because the code is executing through a loop at this point and we don't want to prematurely return out of the method and break the loop

I agree.  Here's a test that fails with `return`, passes with `next`: https://github.com/theRealNG/paper_trail/pull/1